### DISCUSSION TITLE
Assistance needed to use 'oscap-ssh'

### DISCUSSION THREAD

**Discussion by Niba-nazar** [ComplianceAsCode/content#11856] [Q&A]
Hi,

I have installed `OpenSCAP command line tool (oscap) 1.2.16`, on my local machine. I want to scan a remote machine in which I have installed 'oscap' in the similar method I did for my local machine. I have tried the sample `oscap-ssh` command to scan my remote server.

```
oscap-ssh username@IP-address port  xccdf eval --profile xccdf_org.ssgproject.content_profile_standard --results-arf arf.xml --report report.html /path/of/the/scap/SCAP/content/build/ssg-debian10-ds.xml
```
However getting error like this:
```
fish: Unknown command: oscap-ssh
```

I was able to generate the arf.xml and reports.html file for my local machine using `oscap`.

**Comment by Mab879** [comment]
Looks like you don't have the `oscap-ssh` command isn't installed.

I don't have a Debian box to test on but my best guess is that you need to install the `openscap-utils` package.

**Comment by Niba-nazar** [comment]
Hi @Mab879,

Thank you for your response. However, I was unable to locate any information on the `openscap-utils` package in the repository. I also searched through various other sources without success. Could you please provide any documentation or links that could help me learn more about installing the `openscap-utils` package?


In addition, the repository shows that oscap-ssh is included with OpenSCAP version 1.2.3 and later. Therefore, it should function correctly, right?

**Reply by mpurg** [reply]
Based on your original post I assume you are using Debian 10 Buster.

To my knowledge `oscap-ssh` is not available in Debian 10. As mentioned, it is part of the `openscap-utils` package, which appears to be available in Debian 12 Bookworm and later.

A possible solution might be to simply download `oscap-ssh` from the OpenSCAP repository ([link](https://github.com/OpenSCAP/openscap/blob/7f94172ec69cf887b2347f3aff7c17389c629047/utils/oscap-ssh)). I believe that `oscap-ssh` is a standalone bash script, so you might be able to run it without installing anything else.

**Comment by Niba-nazar** [comment]
Hi @mpurg 

>>A possible solution might be to simply download oscap-ssh from the OpenSCAP repository ([link](https://github.com/OpenSCAP/openscap/blob/7f94172ec69cf887b2347f3aff7c17389c629047/utils/oscap-ssh)). I believe that oscap-ssh is a standalone bash script, so you might be able to run it without installing anything else.

The current approach didn't work for me. Are there other alternatives for installing the `oscap-ssh` utility?

**Comment by Niba-nazar** [comment]
Hi, @mpurg 

Now, the oscap-ssh issue mentioned before has been resolved. Unfortunately, as I previously indicated, the operating system is debian11; the problem arises when trying to install requirements on debain Linux servers `apt-get install cmake make libopenscap8 libxml2-utils ninja-build xsltproc`. Here is a copy of the error I'm getting:
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libopenscap8
```

**Comment by mpurg** [comment]
The `libopenscap8` package is not available in debian 11.
You can try to either build the package from source or try to install it from the debian 10 repository, if there are no dependency conflicts.
